src/InvariantTheory/iterators.jl

```diff
@@ -391,6 +391,7 @@ function iterate_basis_linear_algebra(IR::FinGroupInvarRing, d::Int)
 end
 
 Base.eltype(BI::FinGroupInvarRingBasisIterator) = elem_type(polynomial_ring(BI.R))
+# TODO: change this to Base.eltype(::Type{<:FinGroupInvarRingBasisIterator{FinGroupInvarRingT}}) where {FinGroupInvarRingT} = ...
```

Comment:
@joschmitt do you maybe have an idea how to do this more cleanly? I am a bit overwhelmed by the amount of type vars and what to use here

Reply:
Should be possible, I'll have a look tomorrow.

Reply:
Should be resolved.

Reply:
Thanks,!